apis/network/v1alpha3/zone_types.go

```diff
@@ -0,0 +1,130 @@
+// +kubebuilder:resource:scope=Cluster
+type Zone struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ZoneSpec   `json:"spec"`
+	Status ZoneStatus `json:"status,omitempty"`
+}
+
+// ZoneSpec defines the desired state of an AWS Route53 Hosted Zone.
+type ZoneSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ZoneParameters `json:"forProvider"`
+}
+
+// ZoneStatus represents the observed state of a Zone.
+type ZoneStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ZoneObservation `json:"atProvider"`
+}
+
+// ZoneParameters define the desired state of an AWS Route53 Hosted Zone.
+type ZoneParameters struct {
+	// The name of the domain. Specify a fully qualified domain name, for example,
+	// www.example.com. The trailing dot is optional; Amazon Route 53 assumes that
+	// the domain name is fully qualified. This means that Route 53 treats www.example.com
+	// (without a trailing dot) and www.example.com. (with a trailing dot) as identical.
+	//
+	// If you're creating a public hosted zone, this is the name you have registered
+	// with your DNS registrar. If your domain name is registered with a registrar
+	// other than Route 53, change the name servers for your domain to the set of
+	// NameServers that CreateHostedZone returns in DelegationSet.
+	// +immutable
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+
+	// CallerReference is an unique string that identifies the request and that
+	// allows failed Zone create requests to be retried without the risk of
+	// executing the operation twice.
+	// +immutable
+	// +optional
+	CallerReference *string `json:"callerref,omitempty"`
+
+	// Any comments that you want to include about the hosted zone.
+	// +optional
+	Comment *string `json:"comment,omitempty"`
+
+	// A value that indicates whether this is a private hosted zone.
+	// +immutable
+	// +optional
+	PrivateZone *bool `json:"privateZone,omitempty"`
+
+	// (Private hosted zones only) The ID of an Amazon VPC that you're
+	// associating with this hosted zone. You can specify only one Amazon VPC
+	// when you create a private hosted zone.
+	// +immutable
+	// +optional
+	VPCID *string `json:"VPCId,omitempty"`
+
+	// (Private hosted zones only) The region that an Amazon VPC was created in.
+	// +immutable
+	// +optional
+	VPCRegion *string `json:"VPCRegion,omitempty"`
+
+	// (Private hosted zones only) VPCIdRef references a VPC to retrieves its VPC Id
+	// +optional
+	VPCIDRef *runtimev1alpha1.Reference `json:"VPCIdRef,omitempty"`
+
+	// VPCIdSelector selects a reference to a VPC to retrieves its VPC Id
+	// +optional
+	VPCIDSelector *runtimev1alpha1.Selector `json:"VPCIdSelector,omitempty"`
+}
+
+// ZoneObservation keeps the state for the external resource.
+type ZoneObservation struct {
+	// ID is the unique URL representing the new hosted zone.
+	ID string `json:"Id"`
```

Comment:
This is already saved as external-name, so, we don't need to expose it here again.

Reply:
We decided to include it as the final source of truth in case of any discrepancy.